### PR TITLE
test(nix): checks.disko-install for deployments/ovh-test disk-config ⛵

### LINKED ISSUE #55: test: disko VM boot test for deployments/ovh-test disk-config ⛵
## Goal

Verify that `deployments/ovh-test/disk-config.nix` actually produces a bootable system, via the disko library's own test framework (`disko.tests`). The framework handles the QEMU boot dance + fstab + GRUB verification.

Currently we trust that the disko spec is correct because the GPT + ESP + ext4 layout is conventional. But a typo (wrong partition number, missing mountpoint, wrong fstype, mismatched device path) would only surface on first `make install`, costing one cycle of paid OVH hardware time. Catching it pre-cruise in a VM is much cheaper.

## Scope

- [ ] `checks.<system>.disko-boot` exists, uses disko's `disko-tests` framework to boot a VM with `deployments/ovh-test/disk-config.nix` applied.
- [ ] VM successfully boots into NixOS, mounts `/`, `/boot`.
- [ ] Excluded from per-PR CI if the build is too slow (~10-15 min); nightly or opt-in.
- [ ] Documented.

## Open questions

- Is the disko test framework usable directly in `checks.<system>.*`, or does it need its own workflow?
- Does the test need any operator-specific config, or is the disko spec self-contained?

## Out of scope

- Multi-disk layouts (the deployment is single-NVMe; future production deployment is a separate concern)
- LUKS encryption testing (current deployment opts out per #49 lock — production-only concern)

## Refs

For language and reference: **M2 testing milestone** (un-groomed). M3 = production.

Refs #49
Refs #52

### PR DESCRIPTION
**Pull request by Klazomenai**
## Summary

Adds `checks.<system>.disko-install` — a flake check that boots a QEMU VM, formats the virtio disks per `deployments/ovh-test/disk-config.nix`, and runs `nixos-install` (which includes the GRUB install). Catches GPT layout typos, missing ESP / mountpoint wiring, mkfs / fstab errors, and GRUB install failures — bug classes that would otherwise blow up on first `make install` against paid OVH hardware.

Pre-cruise safety net for `#49`. Closes `#55`.

## testBoot = false — decision and rationale

Initial implementation used `testBoot = true` (the framework default) which also boots the installed system and runs post-boot assertions. **The test script PASSED in 421s** — both `mountpoint /` and `mountpoint /boot` succeeded after the booted_machine reached multi-user.target in 2:46. BUT under TCG software emulation (no KVM in the local builder sandbox), the framework's post-test VM teardown then hung in ACPI poweroff for 15+ more minutes before the build artifact could materialise — QEMU CPU dropped to ~7% but the nixos-test-driver kept waiting on the QMP shutdown handshake that never completed cleanly.

`testBoot = false` skips the post-install boot phase entirely. Cuts wallclock from 20+ min (mostly hung in shutdown) to **4.1 min clean**.

| Trade-off | Coverage at `testBoot = false` |
|---|---|
| ✅ Wrong device path | yes (disko fails) |
| ✅ Bad partition table / GPT layout | yes (sgdisk fails) |
| ✅ Bad filesystem type | yes (mkfs fails) |
| ✅ Mountpoint wiring | yes (mount fails) |
| ✅ GRUB **install** failure | yes (`switch-to-configuration boot` runs as part of install phase, AKA the GRUB installer) |
| ❌ GRUB **boot-time** failure | no — planned under `#56`'s full-VM integration; NOT currently covered |
| ❌ post-boot fstab mount | no — planned under `#56`; NOT currently covered |
| ❌ post-boot systemd unit failures | out of scope (this is `#56`'s territory) |

Future revisit: re-enable `testBoot = true` if a CI environment becomes available that offers reliable KVM acceleration for nested-virt nixos tests.

## sgdisk warnings (non-fatal)

The test surfaces two `Could not create partition N from X to Y` sgdisk warnings during the install phase — one each for partition 1 (1 MiB BIOS-boot, EF02) and partition 3 (root, 100% remainder). These are alignment-vs-minimum-size conflicts on the test VM's small disk geometry (`--align-end` + `+1M` / `+0` boundary cases). Disko's fallback path retries with a metadata-only sgdisk invocation that succeeds; the final partition layout is correct (verified via `disko-format` returning success + subsequent mounts working). Real OVH B3 disks are 80+ GiB so the alignment math always succeeds on actual hardware.

## CI policy

`check-full` (push to `main` + nightly cron) runs this automatically via `nix flake check`. NOT added to `check-pr`'s explicit named-check list — the install phase still takes ~4 min under TCG and would meaningfully add to the PR-time budget.

## Test plan

- [x] `nix flake check --no-build` — eval green across 11 checks
- [x] `nix build --no-link .#checks.x86_64-linux.disko-install --print-build-logs` — completes in 4.1 min wallclock (209s test script + 35s VM setup + cleanup)
- [x] Empirical `testBoot = true` verification — both mountpoint assertions ran successfully on the booted VM (421s test script before the framework-shutdown hang) confirming the disk-config IS bootable; the choice to ship `testBoot = false` is purely about CI reliability, not about the disk-config's correctness
- [x] `nix fmt` — formatting clean
- [ ] CI `check-full` passes (will fire on the next push-to-main or nightly cron)

Refs #49
Closes #55

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -485,6 +485,58 @@
...
+        # phase tests pass quickly (~7 min) but the framework's VM
+        # teardown hangs in ACPI poweroff for 15+ min before the
+        # build artifact materialises. The install-phase coverage
+        # is the real safety net for first-cruise OVH provisioning
+        # — boot-time failures (vmlinuz path, initrd, post-boot
+        # fstab mount) are caught by `#56`'s full-VM integration
+        # check, which boots the production configuration end-to-
+        # end. Re-enabling `testBoot = true` here can be revisited
+        # if a future CI environment offers reliable nested-virt
+        # KVM acceleration.
```
[the remark this answers is not in the document]
Comment (Klazomenai) [reply flake.nix]: Fixed in `e2cef03`. The comments asserted present-tense coverage of `#56` (full-VM integration), which is open and unimplemented — a real drift hazard if `#56` never lands. Reworded as forward-looking ("planned under #56; until then, boot-time failures are NOT covered by any automated check and surface only on first real-OVH cycle"). Same rewording applied at both sites in `flake.nix` and the equivalent paragraph in `CONTRIBUTING.md`.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -45,6 +45,20 @@ The check runs as part of `nix flake check` alongside `fmt` and `hardening`, but
 - Real ACME / Let's Encrypt — a self-signed cert is wired directly into the nginx vhost; live HTTP-01 validation against a public DNS name is M3.
 - Performance / load testing.
 
+## Disko boot test
```
[the remark this answers is not in the document]
Comment (Klazomenai) [reply CONTRIBUTING.md]: Good catch — fixed in `e2cef03` via a clean rename across all sites. `checks.disko-boot` → `checks.disko-install`, `name = "ovh-test-disk-boot"` → `name = "ovh-test-disk-install"`, "Disko boot test" section heading → "Disko install test", local-invocation snippet path updated. No stale `disko-boot` references remain in `flake.nix`, `CONTRIBUTING.md`, or the workflow file. Test logic is identical (rename + comment rewrites only) so the 4.1-min wallclock measurement still holds.

File: flake.nix [hunk 3]
```diff
@@ -485,6 +485,70 @@
...
+        # Real OVH B3 instances ship 80+ GiB disks where the
+        # alignment math always succeeds, and they boot UEFI so
+        # the BIOS-boot partition is contingency-only and never
+        # actually consulted.
+        #
+        # CI policy: lives in the flake checks set so `check-full`
+        # (push to `main` + nightly cron) picks it up automatically.
+        # Not added to `check-pr`'s explicit named-check list — the
+        # install phase still takes ~4 min under TCG.
+        checks.disko-install = disko.lib.testLib.makeDiskoTest {
```
[the remark this answers is not in the document]
Comment (Klazomenai) [reply flake.nix:544]: Fixed — PR title updated from `test(nix): checks.disko-boot for deployments/ovh-test disk-config ⛵` to `test(nix): checks.disko-install for deployments/ovh-test disk-config ⛵`, and the PR body updated throughout: `checks.<system>.disko-boot` → `checks.<system>.disko-install`, `nix build .#checks.x86_64-linux.disko-boot` → `nix build .#checks.x86_64-linux.disko-install`. The stale coverage-claim rows in the test-plan table (`#56` boot-time / post-boot fstab) also updated to match the `e2cef03` wording ("planned under `#56`; NOT currently covered") for consistency across code, docs, and PR metadata.
